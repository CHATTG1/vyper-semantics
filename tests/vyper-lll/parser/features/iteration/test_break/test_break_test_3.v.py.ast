%pgm(, ,
  %fdecl(%@public, log, %param(n, %num), %num,
    %assign(%var(c), %decimal(%var(n)))
    %assign(%var(output), 0)
    %forrange(i, 40,
      %if(%compareop(%lt, %var(c), 10),
        %assign(%var(output), %binop(*, %var(i), 10))
        %break)
      %augassign(/=, %var(c), 10))
    %forrange(i, 10,
      %augassign(/=, %var(c), %fixed10(12589000000, 10000000000))
      %if(%compareop(%lt, %var(c), 1),
        %assign(%var(output), %binop(+, %var(output), %var(i)))
        %break))
    %return(%var(output)))
)
