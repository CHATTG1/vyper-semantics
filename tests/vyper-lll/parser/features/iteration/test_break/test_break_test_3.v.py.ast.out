$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 1450352100 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $mstore ( 352 , $mul ( $mload ( 320 ) , 10000000000 ) ) , $mstore ( 384 , 0 ) , $repeat ( 416 , 0 , 40 , $%seq ( $if ( $slt ( $mload ( 352 ) , $mul ( 10 , 10000000000 ) ) , $%seq ( $mstore ( 384 , $clamp ( $mload ( 96 ) , $mul ( $mload ( 416 ) , 10 ) , $mload ( 64 ) ) ) , $break ) ) , $with ( $_mloc , 352 , $mstore ( $_mloc , $clamp ( $mload ( 160 ) , $sdiv ( $mload ( $_mloc ) , $clamp_nonzero ( 10 ) ) , $mload ( 128 ) ) ) ) ) ) , $repeat ( 448 , 0 , 10 , $%seq ( $with ( $_mloc , 352 , $mstore ( $_mloc , $clamp ( $mload ( 160 ) , $with ( $l , $mload ( $_mloc ) , $with ( $r , $clamp_nonzero ( 12589000000 ) , $sdiv ( $mul ( $l , 10000000000 ) , $r ) ) ) , $mload ( 128 ) ) ) ) , $if ( $slt ( $mload ( 352 ) , $mul ( 1 , 10000000000 ) ) , $%seq ( $mstore ( 384 , $clamp ( $mload ( 96 ) , $add ( $mload ( 384 ) , $mload ( 448 ) ) , $mload ( 64 ) ) ) , $break ) ) ) ) , $%seq ( $mstore ( 0 , $mload ( 384 ) ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
